aws/src/test/java/org/apache/iceberg/aws/s3/TestFlakyS3InputStream.java

```diff
@@ -49,10 +51,29 @@
 
 public class TestFlakyS3InputStream extends TestS3InputStream {
 
+  private AtomicInteger resetForRetryCounter;
+
+  @BeforeEach
+  public void setupTest() {
+    resetForRetryCounter = new AtomicInteger(0);
+  }
+
+  @Override
+  S3InputStream newInputStream(S3Client s3Client, S3URI uri) {
+    return new S3InputStream(s3Client, uri) {
+      @Override
+      void resetForRetry() throws IOException {
+        resetForRetryCounter.incrementAndGet();
+        super.resetForRetry();
+      }
```

Comment:
Thanks for fixing this, this was crucial for verifying the retry behavior actually resets the input stream. Seems like we were just getting lucky on the tests before.

Reply:
Yes. That's correct.